### PR TITLE
Disable Towering While Fighting

### PR DESCRIPTION
**Pull request by air-glitch**
I am `disco-dancer` in-game. I would just like to disable towering while fighting because many noobs tower up when I am killing them.

### CODE REVIEW

**Review by moreavy — APPROVED** [review]
Test: Working, Great Job :+1:

**Comment by appgurueu** [comment]
1. This belongs in the place_limit mod IMO
2. Do not call remove node, but set the old node instead. Have a look at how `place_limit` does it. Also, `return true` for the node to not be wasted.

**Comment by moreavy** [comment]
This code disables towering while fighting, but actually, it also disable blocking the way in tunnels too while it shouldn't.

**Comment by air-glitch** [comment]
May I know on which axis (`x`, `y` or `z`) do people tower up?

**Comment by appgurueu** [comment]
> May I know on which axis (`x`, `y` or `z`) do people tower up?

y, of course?

File: mods/pvp/anticoward/init.lua [hunk 1]
```diff
@@ -181,7 +181,7 @@ minetest.register_on_punchplayer(function (player, hitter, time_from_last_punch,
 end)
 
 minetest.register_on_placenode(function(pos, newnode, placer, oldnode, itemstack, pointed_thing)
-	if no_towering[placer:get_player_name()] then
+	if no_towering[placer:get_player_name()] and (placer:get_pos().y = (pos.y + 1)) then
```
Comment (appgurueu) [inline mods/pvp/anticoward/init.lua]: Syntax error, I assumed you intended `==`? Anways, this won't work as expected; you'd have to do a range check for that.

**Review by appgurueu — CHANGES_REQUESTED** [review]

File: mods/pvp/anticoward/init.lua (hunk 1 above)
Comment (air-glitch) [reply mods/pvp/anticoward/init.lua]: I've already fixed `==` bug.

**Comment by moreavy** [comment]
> > May I know on which axis (`x`, `y` or `z`) do people tower up?
> 
> y, of course?

Wrong, `z` axis is the one where players tower up.

File: mods/pvp/anticoward/init.lua [hunk 2]
```diff
@@ -169,3 +169,19 @@ ctf.register_on_new_game(function()
...
+
+minetest.register_on_punchplayer(function (player, hitter, time_from_last_punch, tool_capabilities, dir, damage)
+	no_towering[player:get_player_name()] = true
+	minetest.after(5, function ()
+		no_towering[player:get_player_name()] = false
+	end)
+end)
+
+minetest.register_on_placenode(function(pos, newnode, placer, oldnode, itemstack, pointed_thing)
+	if no_towering[placer:get_player_name()] and (placer:get_pos().y == (pos.y + 1)) then
```
Comment (moreavy) [inline mods/pvp/anticoward/init.lua:184]:
```suggestion
	if no_towering[placer:get_player_name()] and (placer:get_pos().z == pos.z) then
``` I think this should work.

Comment (moreavy) [inline mods/pvp/anticoward/init.lua:184]:
```suggestion
	if no_towering[placer:get_player_name()] and ((math.floor(placer:get_pos().y) == (pos.y + 1)) or (math.ceil(player:get_pos().y) == (pos.y + 1)) then
```

**Pull request closed without merging**